### ISSUE TITLE
better syntactic ux for parallel zql queries

### ISSUE THREAD

**Issue by mccanne** [brimdata/zed#1644]
Currently parallel flowgraphs are specified as semi-colon separated operators `( proc ; proc ; proc) | ...`

This is hard to author and hard to read.  One idea is to highlight the entry point of each parallel path in a group with `=>` as follows...
```
(
  => (
       => count_src=sum(count) by host=id.orig_h | sort host
       => count_dst=sum(count) by host=id.resp_h | sort host
     ) | join host count_dst
  => filter _path=dns qtype_name="TXT" | count_dns=count() by host=id.orig_h | sort host
) | join host count_dns
```
instead of
```
(
  (
    count_src=sum(count) by host=id.orig_h | sort host ;
    count_dst=sum(count) by host=id.resp_h | sort host
  )  | join host count_dst ;
  filter _path=dns qtype_name="TXT" | count_dns=count() by host=id.orig_h | sort host
) | join host count_dns
```

**Comment by mccanne** [comment]
We should also think about how a `switch` operator would jibe with a new syntax like this.

**Comment by henridf** [comment]
> We should also think about how a switch operator would jibe with a new syntax like this.

Over in https://github.com/brimsec/zq/blob/gingest/zng/docs/draft-gingest.md#switched-parallel-flowgraphs, I sketched out 

```
... | switch (
    <boolean expression> => ... | ...;
    <boolean expression> => ... | ...;
    ...
    ) | ...
```

Just a sketch. The reason I had semicolons at the end of each line was for consistency with regular split flowgraph syntax. I wasn't aware of this issue, but if the syntax is changing , then so should this! 

Whatever we go with, the one thing I think we should keep is that the switch condition (`<boolean expression>`) should be syntactically demarcated  from the procs to its right. In my sketch I had started off with `<boolean expression> | ... | ...;` but this is a bit deceptive.
